platform-metrics/src/main/java/org/hypertrace/core/serviceframework/metrics/PlatformMetricsRegistry.java

```diff
@@ -315,6 +369,8 @@ public static synchronized void stop() {
     METRIC_REGISTRY.getNames().forEach(METRIC_REGISTRY::remove);
 
     DEFAULT_TAGS.clear();
+    /* For each meter registry in this composite, it will call the close function */
+    METER_REGISTRY.getRegistries().forEach(MeterRegistry::close);
     METER_REGISTRY.forEachMeter(METER_REGISTRY::remove);
```

Comment:
As part of iterating over meter map, we are also deleting the entry. Can it lead to ConcurrentModException?

Reply:
getRegistries() returns unmodifiable set while remove() is removing from a copy of the unmodifiable set. 
So, they're actually operations on 2 different maps internally.